testsuite/MDAnalysisTests/utils/test_streamio.py

```diff
@@ -174,35 +171,42 @@ def create_NamedStream(self, name=None):
         obj = cStringIO()
         return util.NamedStream(obj, name)
 
-    def test_ospath_funcs(self):
-        ns = self.create_NamedStream()
-
+    @pytest.mark.parametrize('funcname', (
+            "abspath",
+            "basename",
+            "dirname",
+            "expanduser",
+            "normpath",
+            "relpath",
+            "split",
+            "splitext"
+    ))
+    def test_func(self, funcname):
         # - "expandvars" gave Segmentation fault (OS X 10.6, Python 2.7.11 -- orbeckst)
         # - "expanduser" will either return a string if it carried out interpolation
         #   or "will do nothing" and return the NamedStream (see extra test below).
         #   On systems without a user or HOME, it will also do nothing and the test
         #   below will fail.
-        funcs = ("abspath", "basename", "dirname", "expanduser",
-                 "normpath", "relpath", "split", "splitext")
-        def _test_func(funcname, fn=self.textname, ns=ns):
-            func = getattr(os.path, funcname)
-            reference = func(fn)
-            value = func(ns)
-            assert_equal(value, reference,
-                         err_msg=("os.path.{0}() does not work with "
-                                  "NamedStream").format(funcname))
+        ns = self.create_NamedStream()
+        fn = self.textname
+        func = getattr(os.path, funcname)
+        reference = func(fn)
+        value = func(ns)
+        assert_equal(value, reference,
+                     err_msg=("os.path.{0}() does not work with "
+                              "NamedStream").format(funcname))
+
+    def test_join(self, funcname="join", path="/tmp/MDAnalysisTests"):
```

Comment:
We were sending this function "join" explicitly, but the same value is already there as the default parameter, so I skipped adding a parametrize call.